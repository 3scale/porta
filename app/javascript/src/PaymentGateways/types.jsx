// @flow

export type BraintreeFormProps = {
  // eslint-disable-next-line flowtype/no-weak-types
  braintreeClient: any,
  billingAddress?: string,
  threeDSecureEnabled: boolean,
  formActionPath: string,
  countriesList: string,
  selectedCountryCode?: string
}
