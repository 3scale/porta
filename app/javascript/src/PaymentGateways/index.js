// @flow

export { StripeFormWrapper } from './stripe/components/StripeFormWrapper'
export { StripeCardForm } from './stripe/components/StripeCardForm'
export { BraintreeForm } from './braintree/BraintreeForm'
export { BraintreeFormWrapper } from './braintree/BraintreeFormWrapper'
export { BraintreeCardFields } from './braintree/BraintreeCardFields'
export { BraintreeBillingAddressFields } from './braintree/BraintreeBillingAddressFields'
export { BraintreeUserFields } from './braintree/BraintreeUserFields'
export { BraintreeSubmitFields } from './braintree/BraintreeSubmitFields'
export * from './braintree/braintree'
export { Label } from './braintree/components/Label'
export { Input } from './braintree/components/Input'
export { ListItem } from './braintree/components/ListItem'
export * from './braintree/types'
