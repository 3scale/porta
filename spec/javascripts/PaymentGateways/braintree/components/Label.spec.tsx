import React from 'react'
import { mount } from 'enzyme'
import { Label } from 'PaymentGateways'

const props = {
  htmlFor: 'username',
  label: 'Username',
  required: false
}

it('should render properly', () => {
  const wrapper = mount(<Label {...props} />)
  expect(wrapper.exists()).toBe(true)
})

it('should render label', () => {
  const wrapper = mount(<Label {...props} />)
  expect(wrapper.text()).toEqual('Username')
})

it('should add * to mandatory field', () => {
  const propsRequired = { ...props, required: true }
  const wrapper = mount(<Label {...propsRequired} />)
  expect(wrapper.text()).toEqual('Username *')
})
