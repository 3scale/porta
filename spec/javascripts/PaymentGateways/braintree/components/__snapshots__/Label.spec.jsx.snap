// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<Label
  htmlFor="username"
  label="Username"
  required={false}
>
  <label
    className="col-md-4 control-label"
    htmlFor="username"
  >
    Username 
  </label>
</Label>
`;
